File: cindyberg/QLJava/src/org/uva/sea/ql/ast/type/IntegerType.java
package org.uva.sea.ql.ast.type;

import org.uva.sea.ql.ast.TypeVisitor;

public class IntegerType extends Type{
	
	public String toString(){
		return "integer";
	}
	
	@Override
	public boolean isCompatibleWithInteger(){
		return true;
	}
	
	@Override
	public boolean isCompatibleWith(Type t){
		return t.isCompatibleWithInteger();
	}

	@Override
	public <T> T accept(TypeVisitor<T> visitor) {
		return visitor.visit(this);
	}
	

}

File: rperz/QSLib/QSLib/AST/Types/IntegerType.cs
using System.Windows;
using System.Windows.Controls;
using System;
using QSLib.GUI;
using QSLib.Visitors;
namespace QSLib.AST.Types
{
    public class IntegerType : QSType
    {
        public override string Name
        {
            get { return "Number"; }
        }

        public override bool IsCompatible(QSType other)
        {
            if (other == null)
                return false;
            return other.IsInteger();
        }

        public override bool IsInteger()
        {
            return true;
        }

        public override string ToString()
        {
            return "integer";
        }


        public override object Accept<T>(ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override object GetUndefined()
        {
            return 0;
        }
    }
}
